From b75593fd2bb85e79f05972436ef35408cdb83f92 Mon Sep 17 00:00:00 2001
From: Wang ShaoBo <bobo.shaobowang@huawei.com>
Date: Fri, 31 Dec 2021 15:59:49 +0800
Subject: [PATCH] kabi: reserve space for iommu.h
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: b75593fd2bb85e79f05972436ef35408cdb83f92


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4KFY7?from=project-issue
CVE: NA

-------------------------------

reserve space for struct iommu_domain and iommu_ops.

Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Zhen Lei <thunder.leizhen@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/iommu.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/linux/iommu.h b/include/linux/iommu.h
index ed12f5cac0b4..591a6c5d2ddf 100644
--- a/include/linux/iommu.h
+++ b/include/linux/iommu.h
@@ -89,6 +89,10 @@ struct iommu_domain {
 	void *iova_cookie;
 	bool dirty_log_tracking;
 	struct mutex switch_log_lock;
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
 };
 
 enum iommu_cap {
@@ -353,6 +357,14 @@ struct iommu_ops {
 
 	unsigned long pgsize_bitmap;
 	struct module *owner;
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
+	KABI_RESERVE(5)
+	KABI_RESERVE(6)
+	KABI_RESERVE(7)
+	KABI_RESERVE(8)
 };
 
 /**
-- 
2.26.2

